CI/check_size_t.py

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env python3
+
+from pathlib import Path
+import os
+import argparse
+from fnmatch import fnmatch
+import re
+import sys
+
+ex = re.compile(r"(\b(?<!std::)size_t)\b")
+
+github = "GITHUB_ACTIONS" in os.environ
+
+
+def main():
+    p = argparse.ArgumentParser()
+    p.add_argument("input")
+    p.add_argument(
+        "--fix", action="store_true", help="Attempt to fix any license issues found."
+    )
+    p.add_argument("--exclude", "-e", action="append", default=[])
+
+    args = p.parse_args()
+
+    # walk over all files
+    exit = 0
+    for root, _, files in os.walk("."):
+        root = Path(root)
+        for filename in files:
+            # get the full path of the file
+            filepath = root / filename
+            if filepath.suffix not in (".hpp", ".cpp", ".ipp"):
```

Comment:
```suggestion
            if filepath.suffix not in (".hpp", ".h", ".cpp", ".ipp", ".c"):
```
We also have some `.h` and `.c` files using `size_t`. Or should I jsut move the files to the correct extensions?

Reply:
Hmm don't `.c` and `.h` files indicate C files, where `std::size_t` shouldn't be used?

Reply:
In theory yes, but we seem to also have cpp-code inside those files.
I will move them in the next days/week to the "correct" extension :)